network/src/connectivity_manager/mod.rs

```diff
@@ -424,78 +441,85 @@ where
         } else {
             to_connect.len()
         };
+        to_connect
+            .choose_multiple(&mut self.rng, to_connect_size)
+            .map(|(peer_id, peer)| (**peer_id, (*peer).clone()))
```

Comment:
minor nit: does `.cloned()` work?

Reply:
I don't think so, there was definitely a reason I had to do this, and I think it's because it first had to be dereferenced before cloned.